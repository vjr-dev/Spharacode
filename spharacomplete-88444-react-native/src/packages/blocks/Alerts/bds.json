{
  "bds": {
    "name": "Alerts",
    "id": "Alerts",
    "platform": [
      "desktop",
      "mobile"
    ],
    "author": "CMT",
    "summary": null,
    "license": "BSD",
    "version": "1.0.0",
    "pages": [
      {
        "Alerts": {
          "hotspots": [
          ]
        }
      }
    ]
  }
}
