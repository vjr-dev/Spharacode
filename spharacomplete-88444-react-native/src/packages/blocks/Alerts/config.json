{
  "id": "Alerts",
  "verion": "0.0.1",
  "status": "preAlpha",
  "path": "./packages/blocks/Alerts",
  "assembler-adapters": [],
  "mobile": {
    "merge-engine-navigation": {},
    "assembler-exports": [
      "Alerts"
    ],
    "assembler-messages": {
      "output": {
        "btnForgotPassword": [
          "NavigationForgotPasswordMessage"
        ],
        "btnMobileLogIn": [
          "LoginFaliureMessage",
          "LoginSuccessMessage"
        ],
        "btnSocialLogin": [
          "NavigationSocialLogInMessage"
        ],
        "btnEmailLogin": [
          "NavigationEmailLogInMessage"
        ]
      },
      "input": {
        "navigation": [
          "NavigationMobilePhoneLogInMessage"
        ]
      }
    },
    "assembler-dependencies": [],
    "catalog-dependencies": [
      "core",
      "country-code-selector",
      "forgot-password"
    ],
    "assembler-devDependencies": [],
    "merge-engine-screens": {
      "MobileAccountLoginBlock": {
        "file": "MobileAccountLoginBlock.tsx",
        "extends": "MobileAccountLoginController",
        "configURL": "/login/login",
        "validationRulesURL": "api/profile/validation_settings",
        "imports": [
          {
            "value": "CountryCodeSelector",
            "path": "../../../country-code-selector/src/CountryCodeSelector"
          },
          {
            "value": "CustomCheckBox",
            "path": "../../../../components/src/CustomCheckBox"
          },
          {
            "value": "MobileAccountLoginController",
            "path": "./MobileAccountLoginController"
          },
          {
            "value": "{Props}",
            "path": "./MobileAccountLoginController"
          }
        ]
      }
    }
  }
}
