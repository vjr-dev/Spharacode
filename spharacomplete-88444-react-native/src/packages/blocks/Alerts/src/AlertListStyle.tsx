// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { Dimensions, StyleSheet } from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "../../../framework/src/Globals";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export const Styles = StyleSheet.create({
  container: {
    height: sh,
    width: sw,
    flex: 1,
  },
  safeareaView: {
    flex: 1,
    backgroundColor: "#f17234",
  },
  modalView: {
    position: "absolute",
    bottom: 15,
    width: sw - 30,
    alignSelf: "center",
  },
  menuView: {
    width: "100%",
    borderRadius: 15,
    backgroundColor: COLORS.Viewback,
    justifyContent: "center",
    alignItems: "center",
  },
  menuText: {
    padding: 15,
    fontSize: RFValue(15),
    color: COLORS.white,
  },
  seperator: {
    height: 1,
    backgroundColor: COLORS.skipGray,
    width: "90%",
  },
  cancelButton: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    backgroundColor: "#fed99f",
    borderRadius: 25,
  },
  cancelText: {
    paddingVertical: 17,
    fontSize: RFValue(13),
  },
  shadowView: {
    backgroundColor: "#3e3e3e",
    width: sw,
    height: sh,
    position: "absolute",
    opacity: 0.6,
    // zIndex:1,
    flex: 1,
  },
  header: {
    marginLeft: RFValue(10),
    fontSize: RFValue(14),
    color: COLORS.infoGray,
  },
});
// Customizable Area End
