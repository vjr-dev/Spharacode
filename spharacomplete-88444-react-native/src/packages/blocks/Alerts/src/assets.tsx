// Customizable Area Start
export const imgPasswordVisible = require("../assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../assets/ic_password_invisible.png");
export const user = require("../assets/user.png");
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const alert_icon = require("../assets/alert_icon.png");
// Customizable Area End