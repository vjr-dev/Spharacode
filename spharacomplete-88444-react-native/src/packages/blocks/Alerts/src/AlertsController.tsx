// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { IBlock } from "../../../framework/src/IBlock";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  txtInputValue: string;
  txtSavedValue: string;
  enableField: boolean;
  Detail: boolean;
  Chat: boolean;
}

interface SS {
  id: any;
}

export default class AlarmController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.subScribedMessages = [getName(MessageEnum.AccoutLoginSuccess)];

    this.state = {
      txtInputValue: "",
      txtSavedValue: "A",
      enableField: false,
      Detail: true,
      Chat: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
}
// Customizable Area End
