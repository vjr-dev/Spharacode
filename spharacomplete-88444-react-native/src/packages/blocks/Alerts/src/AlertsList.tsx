// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  ActivityIndicator,
  Dimensions,
  Image,
  ImageBackground,
  Modal,
  SafeAreaView,
  ScrollView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import { Styles } from "./AlertListStyle";
import AlertsListController, { Props } from "./AlertsListController";
import * as IMAGE from "./assets";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export default class AlertList extends AlertsListController {
  constructor(props: Props) {
    super(props);
  }
  renderAlerts(item: any, index: number) {
    return (
      <View
        key={it.id}
        style={{
          marginBottom: 10,
          marginTop: 5,
          paddingBottom: 10,
          paddingTop: 10,
          backgroundColor: "#3F4040",
          justifyContent: "center",
        }}
      >
        <TouchableOpacity
          testID="onPressAlert"
          onPress={() =>
            this.props.navigation.navigate("Alerts", {
              date: it.date,
              time: it.time,
              name: it.status_report,
              address: it.address,
              alertType: it.alert_type,
            })
          }
        >
          <View
            style={{
              marginHorizontal: 12,
              flexDirection: "column",
            }}
          >
            <Text
              style={{
                fontSize: 15,
                fontWeight: "600",
                color: COLORS.white,
              }}
            >
              {it.date}
            </Text>
            <View
              style={{
                width: "100%",
                flexDirection: "row",
              }}
            >
              <Text
                style={{
                  marginTop: 5,
                  flex: 1,
                  fontSize: 13,
                  fontWeight: "600",
                  color: COLORS.infoGray,
                }}
              >
                {it.address}
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
  render() {
    return (
      <SafeAreaView style={Styles.safeareaView}>
        <ImageBackground source={IMAGE.back1} style={Styles.container}>
          <ImageBackground source={IMAGE.back2} style={Styles.container}>
            <View
              style={{
                height: hp("7%"),
                width: sw,
                justifyContent: "space-between",
                flexDirection: "row",
                backgroundColor: COLORS.darkorange,
                alignItems: "center",
              }}
            >
              <TouchableOpacity
                testID="alertsListBackButton"
                style={{
                  alignItems: "center",
                  height: "90%",
                  justifyContent: "center",
                  width: "15%",
                }}
                onPress={() => this.props.navigation.pop()}
              >
                <Image
                  source={IMAGE.image_back}
                  style={{
                    height: 15,
                    width: 15,
                    tintColor: COLORS.backgroundGray,
                  }}
                />
              </TouchableOpacity>
              <View
                style={{
                  height: "100%",
                  width: "70%",
                  alignItems: "center",
                  justifyContent: "center",
                }}
              >
                <Text
                  style={{
                    color: COLORS.backgroundGray,
                    fontWeight: "700",
                    fontSize: RFValue(16),
                  }}
                >
                  My Alert
                </Text>
              </View>
              <View style={{ height: "100%", width: "15%" }} />
            </View>

            {this.state.Loader ? (
              <View style={{ flex: 1 }}>
                <Modal visible={this.state.Loader} transparent={true}>
                  <View
                    style={{
                      height: windowHeight,
                      width: windowWidth,
                      alignItems: "center",
                      justifyContent: "center",
                    }}
                  >
                    <View
                      style={{
                        height: 100,
                        width: windowWidth - 20,
                        alignSelf: "center",
                        justifyContent: "center",
                        alignItems: "center",
                      }}
                    >
                      <ActivityIndicator
                        animating={true}
                        size={"large"}
                        color="#f07233"
                      />
                    </View>
                  </View>
                </Modal>
              </View>
            ) : (
              <ScrollView
                style={{
                  width: "100%",
                  marginTop: 15,
                  alignSelf: "center",
                  flex: 1,
                }}
                showsVerticalScrollIndicator={false}
                bounces={false}
              >
                {[Object.keys(this.state.Channels)].map(
                  (item: any, index: number) => {
                    return (
                      <View key={index.toString() + "_outer_id"}>
                        <Text style={Styles.header}>{item}</Text>
                        {this.state.Channels[item]?.map(
                          (it: any, index: number) => {
                            return this.renderAlerts(item, index);
                          }
                        )}
                      </View>
                    );
                  }
                )}
                {/* <SectionList
                  sections={this.state.Channels}
                  keyExtractor={(item, index) => item + index}
                  renderItem={renderItem}
                  renderSectionHeader={({ section: { date } }) => (
                    <Text style={Styles.header}>{date}</Text>
                    // <Text style={Styles.header}>{Obje}</Text>
                  )}
                /> */}
              </ScrollView>
            )}
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}
// Customizable Area End
