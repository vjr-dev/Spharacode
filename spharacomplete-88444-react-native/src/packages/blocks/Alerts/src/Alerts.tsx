// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  Dimensions,
  Image,
  ImageBackground,
  SafeAreaView,
  ScrollView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { RFPercentage, RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import AlertsController, { Props } from "./AlertsController";
import { Styles } from "./AlertsStyle";
import * as IMAGE from "./assets";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export default class Alerts extends AlertsController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <ImageBackground source={IMAGE.back1} style={Styles.container}>
        <ImageBackground source={IMAGE.back2} style={Styles.container}>
          <SafeAreaView>
            <ScrollView style={{ marginBottom: RFPercentage(1), height: sh }}>
              {/* <View
                style={Styles.header_view}
              >
                <TouchableOpacity style={{ marginLeft: 0, height: 60, width: 60, alignItems: "center", justifyContent: "center" }} onPress={() => this.props.navigation.pop()}>
                  <Icon name="chevron-left" color={COLORS.black} size={18} />
                </TouchableOpacity>
                <Text style={Styles.header_txt}>11-08-2020</Text>
              </View> */}
              <View
                style={{
                  height: hp("7%"),
                  width: sw,
                  justifyContent: "space-between",
                  flexDirection: "row",
                  backgroundColor: COLORS.darkorange,
                  alignItems: "center",
                }}
              >
                <TouchableOpacity
                  testID="alertsBackButton"
                  style={{
                    alignItems: "center",
                    height: "90%",
                    justifyContent: "center",
                    width: "15%",
                  }}
                  onPress={() => this.props.navigation.pop()}
                >
                  <Image
                    source={IMAGE.image_back}
                    style={{
                      height: 15,
                      width: 15,
                      tintColor: COLORS.backgroundGray,
                    }}
                  />
                </TouchableOpacity>
                <View
                  style={{
                    height: "100%",
                    width: "70%",
                    alignItems: "center",
                    justifyContent: "center",
                  }}
                >
                  <Text
                    style={{
                      color: COLORS.backgroundGray,
                      fontWeight: "700",
                      fontSize: RFValue(16),
                    }}
                  >
                    {this.props.route.params?.date}
                  </Text>
                </View>
                <View style={{ height: "100%", width: "15%" }} />
              </View>
              <View style={Styles.top_tab}>
                <TouchableOpacity
                  testID="alertsDetailsButton"
                  onPress={() => this.setState({ Detail: true, Chat: false })}
                  style={[
                    Styles.public_view,
                    {
                      borderBottomColor: this.state.Detail
                        ? COLORS.lightyellow
                        : COLORS.backgroundGray,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.public_txt,
                      {
                        color: this.state.Detail
                          ? COLORS.lightyellow
                          : COLORS.skipGray,
                      },
                    ]}
                  >
                    Detail
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  style={[
                    Styles.private_view,
                    {
                      borderBottomColor: this.state.Chat
                        ? COLORS.lightyellow
                        : COLORS.backgroundGray,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.private_txt,
                      {
                        color: this.state.Chat
                          ? COLORS.lightyellow
                          : COLORS.skipGray,
                      },
                    ]}
                  >
                    Chat
                  </Text>
                </TouchableOpacity>
              </View>
              {this.state.Detail ? (
                <View style={{ width: "100%" }}>
                  <View style={Styles.alarm_view}>
                    <Text style={Styles.alram_txt}>Alert detail</Text>
                  </View>
                  <View style={Styles.details_main_view}>
                    <View style={Styles.date_view}>
                      <Text style={{ color: COLORS.infoGray }}>Date :</Text>
                      <Text
                        style={{
                          marginRight: RFPercentage(15),
                          color: COLORS.infoGray,
                          alignSelf: "flex-end",
                        }}
                      >
                        {this.props.route.params?.date}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Time :</Text>
                      <Text style={Styles.info_txt}>
                        {this.props.route.params?.time}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Emergency called # : </Text>
                      <Text
                        style={[
                          Styles.info_txt1,
                          { marginRight: RFPercentage(20.5) },
                        ]}
                      >
                        {this.props.route.params?.alertType}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text
                        style={[
                          Styles.val_txt,
                          { marginRight: RFPercentage(8) },
                        ]}
                      >
                        Location :
                      </Text>
                      <Text style={[Styles.info_txt2, {}]}>
                        {this.props.route.params?.address}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Status report:</Text>
                      <Text
                        style={[
                          Styles.info_txt,
                          { marginRight: RFPercentage(21.5) },
                        ]}
                      >
                        {this.props.route.params?.name}
                      </Text>
                    </View>
                  </View>
                </View>
              ) : (
                <View>
                  <View style={Styles.main_view}>
                    {/* <Icon name="exclamation-triangle" color={COLORS.orange} size={RFValue(31)} style={{ alignSelf: 'center', marginTop: RFValue(10) }} /> */}
                    <Image
                      source={IMAGE.alert_icon}
                      style={{
                        alignSelf: "center",
                        marginTop: RFValue(10),
                        height: RFValue(30),
                        width: RFValue(30),
                      }}
                    ></Image>
                    <Text
                      style={{
                        alignSelf: "center",
                        fontSize: RFValue(14),
                        fontWeight: "500",
                        marginTop: RFValue(4),
                      }}
                    >
                      New Alert!
                    </Text>
                    <View style={Styles.alert_view}>
                      <Text style={Styles.alert_txt}>
                        SPHARA!!! I'm being followed. I'm scared. Send help!
                      </Text>
                      {/* <Text style={Styles.alert_txt}>scared. Send help!</Text> */}
                    </View>
                  </View>
                </View>
              )}
            </ScrollView>
          </SafeAreaView>
        </ImageBackground>
      </ImageBackground>
    );
  }
}
// Customizable Area End
