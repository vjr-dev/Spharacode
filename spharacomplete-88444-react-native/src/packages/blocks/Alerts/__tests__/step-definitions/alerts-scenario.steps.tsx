//@ts-ignore
//@ts-nocheck
import { shallow, ShallowWrapper } from "enzyme";
import { defineFeature, loadFeature } from "jest-cucumber";

import * as helpers from "../../../../framework/src/Helpers";

import React from "react";
import Alerts from "../../src/Alerts";

const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    getParam: jest.fn(),
    navigate: jest.fn(),
    goBack: jest.fn(),
    pop: jest.fn(),
  },
  route:{
    params:{
      date:"date"
    }
  }
};

const feature = loadFeature("./__tests__/features/alerts-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to Alerts Screen", ({ given, when, then }) => {
    let alertsWrapper: ShallowWrapper;
    let instance: Alerts;

    given("I am a User getting alerts", () => {
      alertsWrapper = shallow(<Alerts {...screenProps} />);
      expect(alertsWrapper).toBeTruthy();
      expect(alertsWrapper).toMatchSnapshot();
    });

    when("I navigate to the Alerts Screen", () => {
      instance = alertsWrapper.instance() as Alerts;
      instance.setState({
        txtInputValue: "",
        txtSavedValue: "A",
        enableField: false,
        Detail: true,
        Chat: false,
      });
      expect(alertsWrapper).toBeTruthy();
      expect(alertsWrapper).toMatchSnapshot();
    });

    then("I can press back button", () => {
      let alertsBackButton = alertsWrapper.findWhere(
        (node) => node.prop("testID") === "alertsBackButton"
      );
      alertsBackButton.simulate("press");
      expect(alertsBackButton).toBeTruthy();
    });

    then("I can press details button", () => {
      instance.setState({ Detail: false, Chat: true });
      let alertsDetailsButton = alertsWrapper.findWhere(
        (node) => node.prop("testID") === "alertsDetailsButton"
      );
      alertsDetailsButton.simulate("press");
      expect(alertsDetailsButton).toBeTruthy();
    });
  });
});
