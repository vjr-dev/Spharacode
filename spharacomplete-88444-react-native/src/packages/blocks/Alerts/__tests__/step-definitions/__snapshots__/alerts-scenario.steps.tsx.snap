// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts User navigates to Alerts Screen 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../packages/blocks/Alerts/assets/image_background-1.png",
    }
  }
  style={
    Object {
      "flex": 1,
      "height": 1334,
      "width": 750,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/Alerts/assets/image_background-2.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <SafeAreaView>
      <ScrollViewMock
        style={
          Object {
            "height": 1334,
            "marginBottom": undefined,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255,138,73,1)",
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "space-between",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": "90%",
                "justifyContent": "center",
                "width": "15%",
              }
            }
            testID="alertsBackButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/Alerts/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#404040",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "70%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#404040",
                  "fontSize": undefined,
                  "fontWeight": "700",
                }
              }
            >
              date
            </Text>
          </View>
          <View
            style={
              Object {
                "height": "100%",
                "width": "15%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#404040",
              "flexDirection": "row",
              "height": 53.36,
              "justifyContent": "space-around",
              "marginTop": undefined,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                  "borderBottomWidth": 2,
                  "width": 135,
                },
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                },
              ]
            }
            testID="alertsDetailsButton"
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "rgb(91,91,92)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginBottom": 7,
                  },
                  Object {
                    "color": "rgba(207,185,155,1)",
                  },
                ]
              }
            >
              Detail
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                  "borderBottomWidth": 2,
                  "width": 135,
                },
                Object {
                  "borderBottomColor": "#404040",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(207,185,155,1)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginBottom": 7,
                  },
                  Object {
                    "color": "rgb(91,91,92)",
                  },
                ]
              }
            >
              Chat
            </Text>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": undefined,
                "marginTop": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": undefined,
                  "fontWeight": "600",
                }
              }
            >
              Alert detail
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "space-between",
                "marginLeft": undefined,
                "marginTop": undefined,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                  }
                }
              >
                Date :
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "rgb(163,164,165)",
                    "marginRight": undefined,
                  }
                }
              >
                date
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Time :
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "rgb(163,164,165)",
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Emergency called # : 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(163,164,165)",
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              >
                Location :
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {},
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Status report:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </ScrollViewMock>
    </SafeAreaView>
  </ImageBackground>
</ImageBackground>
`;

exports[`Alerts User navigates to Alerts Screen 2`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../packages/blocks/Alerts/assets/image_background-1.png",
    }
  }
  style={
    Object {
      "flex": 1,
      "height": 1334,
      "width": 750,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/Alerts/assets/image_background-2.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <SafeAreaView>
      <ScrollViewMock
        style={
          Object {
            "height": 1334,
            "marginBottom": undefined,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255,138,73,1)",
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "space-between",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "height": "90%",
                "justifyContent": "center",
                "width": "15%",
              }
            }
            testID="alertsBackButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/Alerts/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#404040",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
                "width": "70%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#404040",
                  "fontSize": undefined,
                  "fontWeight": "700",
                }
              }
            >
              date
            </Text>
          </View>
          <View
            style={
              Object {
                "height": "100%",
                "width": "15%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#404040",
              "flexDirection": "row",
              "height": 53.36,
              "justifyContent": "space-around",
              "marginTop": undefined,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                  "borderBottomWidth": 2,
                  "width": 135,
                },
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                },
              ]
            }
            testID="alertsDetailsButton"
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "rgb(91,91,92)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginBottom": 7,
                  },
                  Object {
                    "color": "rgba(207,185,155,1)",
                  },
                ]
              }
            >
              Detail
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            style={
              Array [
                Object {
                  "borderBottomColor": "rgba(207,185,155,1)",
                  "borderBottomWidth": 2,
                  "width": 135,
                },
                Object {
                  "borderBottomColor": "#404040",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(207,185,155,1)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginBottom": 7,
                  },
                  Object {
                    "color": "rgb(91,91,92)",
                  },
                ]
              }
            >
              Chat
            </Text>
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": undefined,
                "marginTop": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": undefined,
                  "fontWeight": "600",
                }
              }
            >
              Alert detail
            </Text>
          </View>
          <View
            style={
              Object {
                "justifyContent": "space-between",
                "marginLeft": undefined,
                "marginTop": undefined,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                  }
                }
              >
                Date :
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "rgb(163,164,165)",
                    "marginRight": undefined,
                  }
                }
              >
                date
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Time :
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "rgb(163,164,165)",
                    "marginRight": undefined,
                    "marginTop": undefined,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Emergency called # : 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(163,164,165)",
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              >
                Location :
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {},
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "90%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgb(163,164,165)",
                    "marginTop": undefined,
                  }
                }
              >
                Status report:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "flex-end",
                      "color": "rgb(163,164,165)",
                      "marginRight": undefined,
                      "marginTop": undefined,
                    },
                    Object {
                      "marginRight": undefined,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </ScrollViewMock>
    </SafeAreaView>
  </ImageBackground>
</ImageBackground>
`;
